File: code/R/R_Computational_Toolbox_Files/Diffusion_R/diffusion.R
# DIFFUSION new value at cell due to diffusion
diffusion <- function(diffusionRate, site, N, NE, E, SE, S, SW, W, NW){
return ((1 - 8*diffusionRate) * site + diffusionRate*(N+NE+E+SE+S+SW+W+NW))
}

File: products/Introduction_to_Computational_Science_Modules/02_System_Dynamics/MATLAB/Files/Diffusion/diffusion.m
function r = diffusion(diffusionRate, site, N, NE, E, SE, S, SW, W, NW)
% DIFFUSION new value at cell due to diffusion

r = (1 - 8*diffusionRate)*site + diffusionRate*(N+NE+E+SE+S+SW+W+NW);